### PR TITLE
Fixes in power_rev

### PR DESCRIPTION
**Pull request by lbenet**

### CODE REVIEW

**Comment by lbenet** [comment]
Included the Manifest.toml to use 1.0-dev branch in IA

**Comment by lbenet** [comment]
Once https://github.com/JuliaIntervals/IntervalArithmetic.jl/pull/533 is merged (in 1.0-dev branch), all should pass.

**Comment by lbenet** [comment]
The remaining broken tests are some edge cases...

**Comment by lucaferranti** [comment]
with this branch some broken tests are not broken anymore, but even using https://github.com/JuliaIntervals/IntervalArithmetic.jl/pull/533 I get some new failures 🤔

**Comment by lbenet** [comment]
I think https://github.com/JuliaIntervals/IntervalArithmetic.jl/pull/533 branch includes the changes in .lo and .hi (https://github.com/JuliaIntervals/IntervalArithmetic.jl/pull/531), so I would say yes, it includes changes already in 1.0-dev. 

I agree, it seems that it creates some new failures. I think those failures are related to the properties of the implementation that I'm trying to clarify. I don't think those aspects are related to the implementation in IA, but in the way that `pow_rev` is implemented here.

**Comment by lucaferranti** [comment]
hups, I had a small misunderstanding in https://github.com/JuliaIntervals/IntervalArithmetic.jl/pull/533 . In my head I was only thinking / caring about `pow_rev1` and `pow_rev2`, but now I see you were talking about `power_rev` (which tbh I forgot existed).

I'll take another look at this before / after 533 now that I know what I'm talking about :)

File: src/arithmetic.jl [hunk 1]
```diff
@@ -19,7 +19,7 @@ function plus_rev(a::Interval, b::Interval, c::Interval)  # a = b + c
     return a, b_new, c_new
 end
 
-plus_rev(a,b,c) = plus_rev(promote(a,b,c)...)
+plus_rev(a::F, b::F, c) where {T, F<:Interval{T}} = plus_rev(a, b, F(c))
```
Comment (lucaferranti) [inline src/arithmetic.jl:22]: shouldn't this (and the others) be

```julia
function plus_rev(a::Interval{T}, b::Interval{S}, c::Interval(R})
  TSR = promote_type(T, S, R) # not sure if you can do it in one line, but you get the idea
  F = Interval{TSR}
  return plus_rev(F(a), F(b), F(c)
```

but it gets pretty tedious... I get the idea of not having promotion Float64 --> Interval{Float64}, but I think a promotion rule for `Interval{T}, Interval{S}` would be harmless.

In both cases something like `plus_rev(1.0, 1..2, 3..4)` wouldn't work. This can be a bug (if you think in the more ducktyping way) or a feature (if you come from a strong static typing programming language)

**Comment by lbenet** [comment]
> hups, I had a small misunderstanding in [JuliaIntervals/IntervalArithmetic.jl#533](https://github.com/JuliaIntervals/IntervalArithmetic.jl/pull/533) . In my head I was only thinking about `pow_rev1` and `pow_rev2`, but now I see you were talking about `power_rev`.

Sorry, my mistake... I don't remember what was the motivation to make those changes, apart from having a similar coding style than in 1.0-dev... Should I revert those changes, and only concentrate in the `power_rev` function(s) here?

File: src/arithmetic.jl (hunk 1 above)
Comment (lbenet) [reply src/arithmetic.jl:22]: Regarding the second method, which previously had a promotion, I truly don't recall what was the motivation. I guess it was a way of avoiding using `promote`... 

In any case, let me note that many operations in the 1.0-dev branch of IA impose the same `Interval{T}` for the operands: e.g., `(1.0f0 .. 2.f0) - ( 1.5 .. 2.5)` throws a `MethodError` due to ambiguities...

**Comment by lucaferranti** [comment]
no need to revert, it's a good PR and it does fix a few tests (don't know why everything is failing, will investigate soon). Apart for my one comment (which can also be delegated to further PRs) LGTM

File: src/arithmetic.jl (hunk 1 above)
Comment (lucaferranti) [reply src/arithmetic.jl:22]: if it's because of ambiguity, then it's a bug (or at least misleading error message). In general, it's not necessarily wrong to disallow implicit conversions, strongly typed languages do it and for good reasons (see [this accident](https://en.wikipedia.org/wiki/Ariane_flight_V88) )

Comment (lucaferranti) [reply src/arithmetic.jl:22]: but this is probably tangential to this PR

**Comment by dpsanders** [comment]
Sorry for the very late response... What's the status here?
